src/appsignal/helpers.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+import json
+import logging
+from typing import TYPE_CHECKING, Any
+
+from opentelemetry import trace
+
+
+if TYPE_CHECKING:
+    from opentelemetry.trace import Span
+
+logger = logging.getLogger("appsignal")
+
+
+def set_attribute(attribute: str, value: Any, span: Span | None = None) -> None:
+    span = span or trace.get_current_span()
+    # TODO: `trace.get_current_span` will return an INVALID_SPAN
+    # if there is no current span. Should we log something about
+    # it? The Node.js integration does log something about it.
+    #
+    # To detect if it's an invalid span, it seems we would have
+    # to do something like:
+    # `span.get_span_context().is_valid`
+    span.set_attribute(attribute, value)
+
+
+def set_serialised_attribute(
+    attribute: str, value: Any, span: Span | None = None
+) -> None:
+    serialised_value = json.dumps(value)
```

Comment:
You say `Any` is allowed here. Does the JSON dump never fail, even when it's given a Python class or something?

Reply:
Oh, yes, you're right. The dump can fail when the class is not serializable:

```python
class Foo:
    pass

json.dumps(Foo())
# TypeError: Object of type Foo is not JSON serializable
```

And also when the object is circular, which annoyingly, unlike in Node.js, I couldn't find a way to fix:
```
x = {}
x["x"] = x

json.dumps(x)
# ValueError: Circular reference detected
```

Not sure if this can be expressed on the type system, though. I'll have a look. But regardless, we should still account for it at run-time, and probably log a warning instead of letting the error bubble through.